src/ray/object_manager/test/get_request_queue_test.cc

```diff
@@ -0,0 +1,239 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/object_manager/plasma/get_request_queue.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using namespace ray;
+using namespace testing;
+
+/// Test1: AddRequest with object satisfied
+/// Test2: AddRequest with object unsatisfied
+/// Test3: Test Object sealed
+/// Test4: Test Remove
+
+namespace plasma {
+
+class MockClient : public ClientInterface {
+ public:
+  MOCK_METHOD1(SendFd, Status(MEMFD_TYPE));
+  MOCK_METHOD0(GetObjectIDs, const std::unordered_set<ray::ObjectID> &());
+  MOCK_METHOD1(Insert, void(const ObjectID &object_id));
+  MOCK_METHOD1(Remove, void(const ObjectID &object_id));
+};
+
+class MockObjectLifecycleManager : public IObjectLifecycleManager {
+ public:
+  MOCK_METHOD3(CreateObject,
+               std::pair<const LocalObject *, flatbuf::PlasmaError>(
+                   const ray::ObjectInfo &object_info,
+                   plasma::flatbuf::ObjectSource source, bool fallback_allocator));
+  MOCK_CONST_METHOD1(GetObject, const LocalObject *(const ObjectID &object_id));
+  MOCK_METHOD1(SealObject, const LocalObject *(const ObjectID &object_id));
+  MOCK_METHOD1(AbortObject, flatbuf::PlasmaError(const ObjectID &object_id));
+  MOCK_METHOD1(DeleteObject, flatbuf::PlasmaError(const ObjectID &object_id));
+  MOCK_METHOD1(AddReference, bool(const ObjectID &object_id));
+  MOCK_METHOD1(RemoveReference, bool(const ObjectID &object_id));
+};
+
+struct GetRequestQueueTest : public Test {
+ public:
+  GetRequestQueueTest() : io_work_(io_context_) {}
+  void SetUp() override {
+    Test::SetUp();
+    object_id1 = ObjectID::FromRandom();
+    object_id2 = ObjectID::FromRandom();
+    object1.object_info.data_size = 10;
+    object1.object_info.metadata_size = 0;
+    object2.object_info.data_size = 10;
+    object2.object_info.metadata_size = 0;
+    thread_ = std::thread([this] { io_context_.run(); });
```

Comment:
so to get rid of the mutex but still test; you can manually drive io_context_ in the same testing thread by calling 
[io_context_.run_one()](https://www.boost.org/doc/libs/1_75_0/doc/html/boost_asio/reference/io_context/run_one/overload1.html)  So basically it works like following:

```

io_context_.post(some_lambda);
// run_one will execute the some_lambda just posted.
io_context_.run_one();
```

Reply:
It works!